go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -282,8 +282,8 @@ func (b *builtinDateFormat) eval(env *ExpressionEnv) (eval, error) {
 	case *evalTemporal:
 		t = e.toDateTime(datetime.DefaultPrecision, env.now)
 	default:
-		t = evalToDateTime(date, datetime.DefaultPrecision, env.now, env.sqlmode.AllowZeroDate())
-		if t == nil || t.isZero() {
```

Comment:
Did a little cleanup of the zero date handling as well, since it's equivalent to convert not allowing zero dates if we check against `nil` and the zero date anyway.